### PR TITLE
Add 0.11 changes

### PR DESCRIPTION
**Pull request by fmoletta**
This PR contains all the changes introduced by the 0.11 update.
PR list:
#875 Poseidon builtin
#890 Use starknet-crypto for poseidon
#876 ec_op changes
#873 Update keccak builtin
#874 Update layouts

### CODE REVIEW

File: src/hint_processor/builtin_hint_processor/poseidon_utils.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+) -> Result<(), HintError> {
+    let n = get_integer_from_var_name("n", vm, ids_data, ap_tracking)?;
+
+    let value = if n.as_ref() >= &Felt252::from(10) {
+        Felt252::one()
+    } else {
+        Felt252::zero()
+    };
```
[the remark this answers is not in the document]
Comment (fmoletta) [reply src/hint_processor/builtin_hint_processor/poseidon_utils.rs]: [0921e78]

File: src/hint_processor/builtin_hint_processor/poseidon_utils.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+) -> Result<(), HintError> {
+    let n = get_integer_from_var_name("n", vm, ids_data, ap_tracking)?;
+
+    let value = if n.as_ref() >= &Felt252::from(2) {
+        Felt252::one()
+    } else {
+        Felt252::zero()
+    };
```
[the remark this answers is not in the document]
Comment (fmoletta) [reply src/hint_processor/builtin_hint_processor/poseidon_utils.rs]: [0921e78]

File: src/vm/security.rs [hunk 3]
```diff
@@ -47,14 +48,15 @@ pub fn verify_secure_runner(
...
+    let program_segment_size = program_segment_size.unwrap_or(runner.program.data.len());
+    let program_lenght = vm
         .segments
         .memory
         .data
         .get(program_segment_index)
         .map(|segment| segment.len());
     // + 1 here accounts for maximum segment offset being segment.len() -1
-    if program_segment_size >= Some(runner.program.data.len() + 1) {
+    if program_lenght >= Some(program_segment_size + 1) {
```
Comment (Oppen) [inline src/vm/security.rs:59]: Typo:
```suggestion
    if program_length >= Some(program_segment_size + 1) {
```

**Review by Oppen — COMMENTED** [review]
Seems OK. Just cover the lines reported by codecov and a benchmark program for Poseidon would be greatly appreciated.

**Review by juanbono — APPROVED** [review]

**Review by Oppen — APPROVED** [review]

**Pull request merged**